keylime/registrar_common.py

```diff
@@ -68,60 +104,70 @@ def do_GET(self):
         agent to be returned. If the agent_id is not found, a 404 response is returned.
         """
+        session = SessionManager().make_session(engine)
         rest_params = common.get_restful_params(self.path)
         if rest_params is None:
-            common.echo_json_response(self, 405, "Not Implemented: Use /agents/ interface")
+            common.echo_json_response(
+                self, 405, "Not Implemented: Use /agents/ interface")
             return
 
         if "agents" not in rest_params:
             common.echo_json_response(self, 400, "uri not supported")
-            logger.warning('GET returning 400 response. uri not supported: ' + self.path)
+            logger.warning(
+                'GET returning 400 response. uri not supported: ' + self.path)
             return
 
         agent_id = rest_params["agents"]
 
         if agent_id is not None:
-            agent = self.server.db.get_agent(agent_id)
+            try:
+                agent = session.query(RegistrarMain).filter_by(agent_id=agent_id).first()
+            except SQLAlchemyError as e:
+                logger.error(f'SQLAlchemy Error: {e}')
 
             if agent is None:
                 common.echo_json_response(self, 404, "agent_id not found")
-                logger.warning('GET returning 404 response. agent_id ' + agent_id + ' not found.')
+                logger.warning(
+                    'GET returning 404 response. agent_id ' + agent_id + ' not found.')
                 return
 
-            if not agent['active']:
+            if not agent.active:
                 common.echo_json_response(self, 404, "agent_id not yet active")
-                logger.warning('GET returning 404 response. agent_id ' + agent_id + ' not yet active.')
+                logger.warning(
+                    'GET returning 404 response. agent_id ' + agent_id + ' not yet active.')
                 return
 
             response = {
-                'aik': agent['aik'],
-                'ek': agent['ek'],
-                'ekcert': agent['ekcert'],
-                'regcount': agent['regcount'],
+                'aik': agent.aik,
+                'ek': agent.ek,
+                'ekcert': agent.ekcert,
+                'regcount': agent.regcount,
             }
 
-            if agent['virtual']:
-                response['provider_keys']= agent['provider_keys']
+            if agent.virtual:
+                response['provider_keys'] = agent.provider_keys
 
             common.echo_json_response(self, 200, "Success", response)
             logger.info('GET returning 200 response for agent_id:' + agent_id)
         else:
             # return the available registered uuids from the DB
-            json_response = self.server.db.get_agent_ids()
-            common.echo_json_response(self, 200, "Success", {'uuids':json_response})
+            json_response = session.query(RegistrarMain.agent_id).all()
```

Comment:
Note that this breaks our API, since this now returns a dictionary instead of a list, e.g., 
````javascript
[{'agent_id': 'XXXX'}]
````
vs. 

````javascript
['XXXX']
````

We should check to make sure this isn't also the case in other areas of the code, this is the first one I stumbled across.

Reply:
ok, interesting..this is why I used literal.eval for a similar query used in the verifier. I will try this out.

Reply:
How does this get called? I have set some type debug statements but can't replicate the conditions to make this part of the code get called?

Reply:
I noticed it because the WebApp uses this API to get information about the registered agents in the system.  The WebApp expects to be getting an array of UUIDs back (not an array of dictionaries), so the WebApp is failing right now.

Reply:
understood, thanks @jetwhiz trying to debug this now.

Reply:
Just for reference, the actual error is occurring here: https://github.com/keylime/keylime/blob/6068e1023de334d46b13661a093d2c5b31592f07/keylime/tenant_webapp.py#L387-L389

The `agent` variable used to be a string, but is now a dictionary (resulting in an exception when trying to perform `agents[agent]`. 

Also note that the `one_or_none()` issue mentioned in a previous review also is breaking the WebApp, so that needs to be fixed as well before you make it to this issue.

Reply:
could you try this if you already have an environment:
```
import ast
for agent in agent_list: 
     ast.literal_eval(agents[agent]) = await self.get_agent_state(agent_id) 
```

Reply:
got the `one_or_none()` issue queued up.

Reply:
I can't even get it to install when doing that, it gives `SyntaxError: can't assign to function call`

Reply:
Sorry, we might need another approach. literal_eval is more useful for converting to a dict. Just doing some port forwards to allow debug env.,